WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -121,10 +123,19 @@ public void onMessageReceived(String from, Bundle data) {
             return;
         }
 
-        // Handle helpshift PNs
-        if (TextUtils.equals(data.getString("origin"), "helpshift")) {
-            // TODO: Handle Zendesk PNs instead
-//            HelpshiftHelper.getInstance().handlePush(this, new Intent().putExtras(data));
+        // Handle Zendesk PNs
+        // TODO: Move the contents of this to a helper
+        if (TextUtils.equals(data.getString("type"), "zendesk")) {
```

Comment:
Where's this  `type`  and  `zendesk` value coming from? I'm sure you checked but I couldn't find it in the docs, can you link to it or was it identified empirically?

Reply:
It's coming from our own servers. Zendesk don't send push notifications, they ping our servers and our servers then send it through. I'll add constants for those when everything is done.

Reply:
Ah! gotcha. All good then, I just wanted to point out that we should make sure to have control over the PN->App cycle entirely, and you just confirmed so.